lib/ntpUtil.js

```diff
@@ -29,6 +57,62 @@ const generatePublicKeyAndID = (privateKeyFile) => {
   }
 }
 
+const isSupportedSchemaVersion = (version) => {
+  return version === 1 || version === 2;
+}
+
+const prepareAssets = (jsonFileUrl, targetResourceDir, targetJsonFileName) => {
+  return new Promise(function (resolve, reject) {
+    let jsonFileBody = '{}'
+
+    // Download and parse jsonFileUrl.
+    // If it doesn't exist, create with empty object.
+    request(jsonFileUrl, async function (error, response, body) {
+      if (error) {
+        console.error(`Error from ${jsonFileUrl}:`, error)
+        return reject(error)
+      }
+      if (response && response.statusCode === 200) {
+        jsonFileBody = body
+      }
+      let photoData = {}
+      try {
+        photoData = JSON.parse(jsonFileBody)
+      } catch (err) {
+        console.error(`Invalid json file ${jsonFileUrl}`)
+        return reject(error)
```

Comment:
Does the error get written to stdout when reject is called? Would be important to have a way to see the raw error when debugging

Reply:
Will test about this.

Reply:
You're right. error msg is not written when rejected. need to find how to print :)